File: Python/day1.part2.py
import re
def convert_word_to_digit(word):
    word_to_digit_mapping = {
        'one': 1,
        'two': 2,
        'three': 3,
        'four': 4,
        'five': 5,
        'six': 6,
        'seven': 7,
        'eight': 8,
        'nine': 9
    }
    return word_to_digit_mapping.get(word)
    
def extract_number(line):
      pattern = r'(?=(one|two|three|four|five|six|seven|eight|nine|\d))'
      regex = re.compile(pattern)
      get_numbers = re.findall(regex,line)
      
      first_element = convert_word_to_digit(get_numbers[0]) if not get_numbers[0].isdigit() else get_numbers[0]
      last_element = convert_word_to_digit(get_numbers[-1]) if not get_numbers[-1].isdigit() else get_numbers[-1]
      
      return int(f"{first_element}{last_element}")
 
def calculate_calibration(file_path):
    calibration = 0
    with open (file_path,"r") as file:
        lines = file.readlines()
        for line in lines:
            extracted_number = extract_number(line)
            calibration += extracted_number
            
    return calibration

file_path = "input/inputday1.txt"
print(f"Sum of Combined Calibration Values: {calculate_calibration(file_path)}")

File: JavaScript/2023/day1part2.js
const fs = require('fs');

function calculateCalibration(lines) {
    let sum = 0;
const numbers = [ "zero", "one","two","three","four", "five","six","seven", "eight", "nine"]
lines.forEach(line => {
let firstDigit = line.match(/one|two|three|four|five|six|seven|eight|nine|\d/)[0]
let lastDigit = [...line.matchAll(/(?=(\d|one|two|three|four|five|six|seven|eight|nine))/g)].slice(-1)[0][1]
if (isNaN(lastDigit)) lastDigit = numbers.indexOf(lastDigit)
if (isNaN(firstDigit)) firstDigit = numbers.indexOf(firstDigit)

if (firstDigit && lastDigit) {
    
    const fullNumber = firstDigit.toString() +lastDigit.toString()
    
    sum += +fullNumber
  }
});

return sum;
}



 
  fs.readFile('input.txt', 'utf8', (err, data) => {
    if (err) {
      console.error('Error reading the file:', err);
      return;
    }
  
 
    const lines = data.split('\n');
  
    const result = calculateCalibration(lines);
    console.log("Total calibration value:", result);
  });